docs/quickstart.md

```diff
@@ -81,147 +36,6 @@ Run the following script if you want to destroy your local Armada cluster:
 ./docs/local/destroy.sh
 ```
 
-### Manual setup
```

Comment:
Since these have been tested to work, can they be moved to another document for manual setup, for folks (like me) who wouldn't be very comfortable running a bash script to configure an open source project.

Reply:
Can't you just run the script step by step in that case? Most projects I see use 'pip install foo' or 'gem install bar'. I'd like to get out of the business of maintaining both versions.

Reply:
https://github.com/G-Research/armada/discussions/1109

I like to follow similar patterns in the open source community.  I have not found any example that tells someone to run a bash script.   I think we should actually scrap the script and have code blocks that people can copy.  I would also prefer us to clean up this installation process so it is possible to do in a few steps rather than 10-20.

In the CNCF projects, I see a common pattern is to use kubectl to install a project onto an existing cluster.   

It may seem weird but it is common practice to have people copy code blocks into their terminal to run for these quick starts.  Look at the airflow documentation in my discussion.  

I have maybe a 10-15% confidence that this script will actually work on people's machines.  That is a big part why I wrote the discussion because I think we should adopt the changes I proposed in that discussion before we update the quickstart.  

So far, Pulsar seems to be a big issue on most of our machines so this script will probably not work for most of our users.  I think its related to the fickleness of Kind/Minikube as a local method for deploying Kubernetes.  Deploying via helm charts is typically reserved for a production installation and the charts don't seem to work well with local installation.  

Maybe we should have a discussion about this before we update the quickstart?

Reply:
In terms of maintaining , I agree.  But no developer on our team or in GR uses the quickstart guide as a means of developing.  So it will always be out of date.  

I would prefer that we migrate our quickstart guide to be focused on a developer persona with using docker-compose which is a common way that we develop our code.  We also keep this up to date.  My hope is to also use docker-compose environment in our CI as a way to keep our dev section up to date.  If not, I find this to be the most reliable way of setting up Armada for development so I think it would be kept update.   

I would like to use a production ready K8 cluster in our quickstart guide because I think if we release this as is, we will spend a lot of time debugging issues that people have on various machines.  We already have a series of issues about how our original quickstart guide doesn't work on certain machines.    

This script currently doesn't work on mac M1, AWS EC2, or Rich's linux box.  It may work on other teams machines if they have enough memory.  But I think this script will cause more issues than it is worth IMO.

Reply:
This script just worked fine on my Mac M1, including the Pulsar bits. How about we take this as a cleanup on the page and enhance it further when we have a K8 cluster for new users to try out? I think this is still something of a win, if not the end state.

Reply:
@kannon92 @richscott @jayofdoom @rankindaveyg 
A lot of CNCF projects can only be installed in Kubernetes and cannot possibly work outside of k8s. (Argo, Istio, Cert Manager...)

Armada has a hard dependency on Kubernetes, where the executors need a k8s cluster in which to schedule a job, but rest of components can be run outside of k8s. Even tho some components can run outside of k8s, it is not their natural habitat. Natural habitat for Armada and its components is inside a k8s cluster.

My proposal is we redo the Quickstart and move towards using the Armada Bundle helm chart to install a fully functional Armada cluster.

Guide could have a prerequisite for a k8s cluster, and if the user does not have one, we provide docs to setup one (minikube or kind), and after that we just tell them now use `helm install armada-bundle gresearch/armada-bundle`.

WDYT?

Reply:
Wow.  I haven't got this script working on either of my machines..

Reply:
I think that is good but the main point of using Armada is to provide multi cluster scheduling.  

Will this be handled by a helm chart bundle?  Or will that install the executor onto the same cluster as the server?

Reply:
The control plane/server/main cluster part has the most components, you need cert manager, prometheus, nginx controller... where the executor cluster only needs one thing, the executor.

Also, Quickstart guides for popular tools assume you have a k8s cluster and dive straight into installing using kubectl, helm, kustomize...

* ArgoCD - https://argo-cd.readthedocs.io/en/stable/getting_started/
* Cert Manager - https://cert-manager.io/docs/installation/
* Istio (they provide a bash script but all whom I knew used Istio, absolutely avoided the bash script and went straight to Helm chart) - https://istio.io/latest/docs/setup/getting-started/
* NGINX Controller (it has nuances between different k8s flavours, so it has examples for varios k8s deployments) - https://kubernetes.github.io/ingress-nginx/deploy/

Reply:
Holy moly, we are creeping the scope of what is just a rearranging of the furniture here. I'd like to propose that we accept this work as-is as a cleanup of the website and move discussions of the future of the developer experience into a new discussion.

Reply:
It sucks that K8S (and K8S lookalikes such as Kind/minikube/k3s) is such a buggy mess to get fully installed and running correctly. I'm seriously drawn towards just telling potential Armada users "first, get a running K8S cluster" (they may have one in-house already, or we suggest using AWS EKS or Google's managed clusters), then we provide a Helm chart to install Armada. 

But if it runs at least reasonably well using Kind as the cluster manager/k8s-simulator on Docker Desktop for Mac, can/should we declare that as the supported dev-desktop configuration for now?

Reply:
Our point is that most OSS do not tell you to run a script from their webpage.  If they do, you use something like `pip install` or `helm install` which is verified.  I think that running a script without knowing what's inside is a security issue in most organizations.  Its security 101 to not run an executable from someone you don't know sent to you via email.  I think the same thing should apply on any website.  
 
 The other issue is that this script doesn't really provide a stable environment.  So we "fix" the issue on the webpage but we will keep fixing support issues for people who try to run this as an evaluation.  And it makes us look sloppy and we don't get any people adopting our software.

Reply:
For what it's worth, I just noticed that the Volcano project essentially expects you to have a running K8S cluster as a prequisite:  https://volcano.sh/en/docs/installation/  - then guide to installing Volcano via `kubectl` with YAML spec files, or using `helm` with their public charts.